src/Test/Utilities/Portable/Metadata/IlasmUtilities.cs

```diff
@@ -29,7 +29,13 @@ private static string GetIlasmPath()
             }
             else
             {
-                return Path.Combine(AppContext.BaseDirectory, "ilasm");
+                var directory = AppContext.BaseDirectory;
+                string path = null;
+                while (directory != null && !File.Exists(path = Path.Combine(directory, "Binaries", "Tools", "ilasm")))
+                {
+                    directory = Path.GetDirectoryName(directory);
+                }
+                return directory == null ? null : path;
```

Comment:
In the case of `directory == null` we should throw an exception. Everything is busted at that point. Should have a declarative error message stating what went wrong.